agent/xds/listeners.go

```diff
@@ -264,8 +264,9 @@ func (s *ResourceGenerator) listenersFromSnapshotConnectProxy(cfgSnap *proxycfg.
 		}
 		outboundListener.ListenerFilters = append(outboundListener.ListenerFilters, tlsInspector)
 	}
-	// Looping over explicit upstreams is only needed for cross-peer because
-	// they do not have discovery chains.
+
+	// Looping over explicit and implicit upstreams is only needed for cross-peer
+	// because they do not have discovery chains.
 	for _, uid := range cfgSnap.ConnectProxy.PeeredUpstreamIDs() {
```

Comment:
Why is this not using `PeeredUpstreamIDsFromConfig` like the clusters and endpoints loops?

Reply:
`UpstreamConfig` doesn't contain the implicit peered upstreams. So `PeeredUpstreamIDs` is explicit + implicit upstreams (which mirrors disco chains in that they contain both implicit and explicit upstreams)

Reply:
All resources use `PeeredUpstreamIDs` and the `FromConfig` variant has been removed